src/realm/object_converter.cpp

```diff
@@ -371,17 +369,17 @@ int InterRealmValueConverter::cmp_src_to_dst(Mixed src, Mixed dst, ConversionRes
         }
         else {
             Obj dst_link;
-            if (m_opposite_of_src->get_primary_key_column()) {
-                Mixed src_link_pk = m_opposite_of_src->get_primary_key(src_link_key);
-                dst_link = m_opposite_of_dst->create_object_with_primary_key(src_link_pk, did_update_out);
+            if (m_opposite_of_dst == m_opposite_of_src) {
+                // if this is the same Realm, we can use the ObjKey
+                dst_link = m_opposite_of_dst->get_object(src_link_key);
```

Comment:
The previous order of these checks resulted in a pointless PK lookup when copying objects in the same table.